scripts/tf-tests/run-jobs.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python3
+
+import re
+import argparse
+import subprocess
+import json
+import logging
+from time import sleep
+from pathlib import Path
+from typing import Dict, Optional
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(
+        description="Submit jobs and monitor their status on Testflinger."
+    )
+    parser.add_argument(
+        "--machines-file",
+        type=Path,
+        default="machines.txt",
+        help="Path to the file with machines Canonical IDs",
+    )
+    parser.add_argument(
+        "--template-file",
+        type=Path,
+        default="tf-job.yaml",
+        help="Path to Testflinger job template file",
+    )
+    parser.add_argument(
+        "--output-dir",
+        type=Path,
+        default="job_outputs",
+        help="Path to job outputs directory",
+    )
+    parser.add_argument(
+        "--poll-interval",
+        type=int,
+        default=30,
+        help="Time delay between status checks in seconds",
+    )
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--send-jobs",
+        action="store_true",
+        help="Only submit jobs without monitoring their statuses",
+    )
+    group.add_argument(
+        "--check-status",
+        action="store_true",
+        help="Only check job statuses without submitting new jobs",
+    )
+    return parser.parse_args()
+
+
+# Job Submission Functions
+def load_canonical_ids(filename: Path) -> list:
+    """Reads the Canonical IDs from a file."""
+    with open(filename, "r", encoding="utf8") as file:
+        return file.read().strip().splitlines()
+
+
+def create_job_yaml(template_file: Path, canonical_id: str) -> str:
+    """Creates a modified job YAML for a specific Canonical ID."""
+    with open(template_file, "r", encoding="utf8") as file:
+        job_yaml = file.read()
+    return job_yaml.replace("$CANONICAL_ID", canonical_id)
+
+
+def write_temp_job_file(job_yaml: str, output_dir: Path, canonical_id: str) -> Path:
+    """Writes the modified job YAML to a temporary file."""
+    temp_job_file = output_dir / f"{canonical_id}_tf-job.yaml"
+    temp_job_file.write_text(job_yaml)
+    return temp_job_file
```

Comment:
This is not major but the split in two functions here seems artificial. You can even see that in the docstring for `write_temp_job_file`: it is not easy to describe what this function does without referring to the output of the other function, i.e. "the modified job YAML". I would suggest merging these two into one function: what you are really doing here is instantiating the template with a value for `$CANONICAL_ID`.

I would also suggest accepting the path for the output (instantiated) file as an argument, rather than hardcoding the `temp_job_file` in the function itself. So the signature for the merged function would be something like:
```
def instantiate_job_yaml(template: Path, output: Path, canonical_id: str):
```